### PR TITLE
Support for feat/rosetta

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Support `?withLogs=true` on blocks & hyperblocks
 - Support account query parameter `?onFinalBlock=true`; return `blockInfo` in response
 - Handle `transaction.IsRefund` property
 - Handle `transaction.InitiallyPaidFee` property
 - Move **rosetta API** implementation to a separate repository: https://github.com/ElrondNetwork/rosetta
 - New fields on miniblocks: `processingType`, `constructionType`, `isFromReceiptsStorage`.

### CODE REVIEW

File: data/account.go [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package data
 
-// Account defines the data structure for an account
+// AccountModel defines an account model (with associated information)
+type AccountModel struct {
+	Account   Account
+	BlockInfo BlockInfo
+}
+
+// AccountResource defines the data structure for an account
```
Comment (bogdan-rosianu) [inline data/account.go:9]: mismatch between struct name and comment

File: api/groups/baseAccountsGroup.go [hunk 2]
```diff
@@ -48,14 +48,23 @@ func NewAccountsGroup(facadeHandler data.FacadeHandler) (*accountsGroup, error)
 	return ag, nil
 }
 
-func (group *accountsGroup) getAccountFromFacade(c *gin.Context) (*data.Account, int, error) {
-	addr := c.Param("address")
-	acc, err := group.facade.GetAccount(addr)
+func (group *accountsGroup) respondWithAccount(c *gin.Context, transform func(*data.AccountModel) gin.H) {
+	address := c.Param("address")
+
+	options, err := parseAccountQueryOptions(c)
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply api/groups/baseAccountsGroup.go:54]: Added.

File: common/queries.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+const (
+	// UrlParameterWithTransactions represents the name of an URL parameter
+	UrlParameterWithTransactions = "withTxs"
+	// UrlParameterWithLogs represents the name of an URL parameter
+	UrlParameterWithLogs = "withLogs"
+	// UrlParameterOnFinalBlock represents the name of an URL parameter
+	UrlParameterOnFinalBlock = "onFinalBlock"
+	// UrlParameterOnStartOfEpoch represents the name of an URL parameter
+	UrlParameterOnStartOfEpoch = "onStartOfEpoch"
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply common/queries.go:16]: Moved.

File: data/account.go (hunk 1 above)
Comment (andreibancioiu) [reply data/account.go:9]: Fixed.

**Review by bogdan-rosianu — COMMENTED** [review]
only the findings from Mihai

File: common/options.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+// AccountQueryOptions holds options for account queries
+type AccountQueryOptions struct {
+	OnFinalBlock   bool
+	OnStartOfEpoch uint32
+}
+
+// BuildUrlWithAccountQueryOptions builds an URL with block query parameters
+func BuildUrlWithAccountQueryOptions(path string, options AccountQueryOptions) string {
+	url := url.URL{Path: path}
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply common/options.go:68]: Fixed.

File: common/options.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+// TransactionSimulationOptions holds options for transaction simulation requests
+type TransactionSimulationOptions struct {
+	CheckSignature bool
+}
+
+// BuildUrlWithBlockQueryOptions builds an URL with block query parameters
+func BuildUrlWithBlockQueryOptions(path string, options BlockQueryOptions) string {
+	url := url.URL{Path: path}
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply common/options.go:46]: Fixed.

File: data/account.go [hunk 6]
```diff
@@ -1,5 +1,11 @@
 package data
 
+// AccountModel defines an account model (with associated information)
+type AccountModel struct {
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply data/account.go:4]: Fixed.

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by miiu96 — DISMISSED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Pull request merged**